source/request-response/request_response_client.c

```diff
@@ -397,3 +806,390 @@ struct aws_mqtt_request_response_client *aws_mqtt_request_response_client_releas
+        AWS_OP_SUCCESS);
+    AWS_FATAL_ASSERT(
+        aws_byte_buf_append_and_update(&storage->operation_data, &storage->options.correlation_token) ==
+        AWS_OP_SUCCESS);
+    AWS_FATAL_ASSERT(
+        aws_byte_buf_append_and_update(&storage->operation_data, &storage->options.subscription_topic_filter) ==
+        AWS_OP_SUCCESS);
+
+    for (size_t i = 0; i < request_options->response_path_count; ++i) {
+        struct aws_mqtt_request_operation_response_path response_path = request_options->response_paths[i];
+
+        AWS_FATAL_ASSERT(
+            aws_byte_buf_append_and_update(&storage->operation_data, &response_path.topic) == AWS_OP_SUCCESS);
+        AWS_FATAL_ASSERT(
+            aws_byte_buf_append_and_update(&storage->operation_data, &response_path.correlation_token_json_path) ==
+            AWS_OP_SUCCESS);
+
+        aws_array_list_push_back(&storage->operation_response_paths, &response_path);
+    }
+
+    storage->options.response_paths = storage->operation_response_paths.data;
+}
+
+int aws_mqtt_request_response_client_submit_request(
+    struct aws_mqtt_request_response_client *client,
+    const struct aws_mqtt_request_operation_options *request_options) {
+
+    if (client == NULL) {
+        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+    }
+
+    if (!s_are_request_operation_options_valid(client, request_options)) {
+        /* all failure cases have logged the problem already */
+        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+    }
+
+    struct aws_allocator *allocator = client->allocator;
+    struct aws_mqtt_rr_client_operation *operation =
+        aws_mem_calloc(allocator, 1, sizeof(struct aws_mqtt_rr_client_operation));
+    operation->allocator = allocator;
+    operation->type = AWS_MRROT_REQUEST;
+
+    s_aws_mqtt_request_operation_storage_init_from_options(
+        &operation->storage.request_storage, allocator, request_options);
+    s_aws_mqtt_rr_client_operation_init_shared(operation, client);
+
+    aws_event_loop_schedule_task_now(client->loop, &operation->submit_task);
+
+    return AWS_OP_SUCCESS;
+}
+
+void s_aws_mqtt_streaming_operation_storage_init_from_options(
+    struct aws_mqtt_streaming_operation_storage *storage,
+    struct aws_allocator *allocator,
+    const struct aws_mqtt_streaming_operation_options *streaming_options) {
+    size_t bytes_needed = streaming_options->topic_filter.len;
+
+    storage->options = *streaming_options;
+    aws_byte_buf_init(&storage->operation_data, allocator, bytes_needed);
+
+    AWS_FATAL_ASSERT(
+        aws_byte_buf_append_and_update(&storage->operation_data, &storage->options.topic_filter) == AWS_OP_SUCCESS);
+}
+
+struct aws_mqtt_rr_client_operation *aws_mqtt_request_response_client_create_streaming_operation(
+    struct aws_mqtt_request_response_client *client,
+    const struct aws_mqtt_streaming_operation_options *streaming_options) {
+
+    if (client == NULL) {
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
```

Comment:
Trivial: same as above. Should we log the client null is the invalid argument?

Reply:
Same response